### ISSUE TITLE
Windows defender found a trojan in ChatGPT.exe[Security]

### ISSUE THREAD

**Issue by KullAxel** [lencx/ChatGPT#440]
### Description

After installing the program windows detected a trojan in the ChatGPT.exe files.

Trojan:Win64/Malgent!MSR

Anyone else had this happaning?

**Comment by joshuafuller** [comment]
Windows Defender just flagged the app for me as well.

**Comment by fant5y** [comment]
Did that to me too, 10 Minutes ago.

Threat quarantined
18.02.2023 19:28
Detected: Trojan:Win64/MaIgent!MSR
Status: Quarantined
Quarantined files are in a restricted area where they can't harm your device. They will
be removed automatically.
Date: 18.02.2023 19:29
Details: This program is dangerous and executes commands from an attacker.
Affected items:
file: C:\Program Files\ChatGPT\ChatGPT.exe
file: C:\ProgramData\Microsoft\Windows\Start Menu\Programs\ChatGPT
\ChatGPT.lnk
file: C:\Users\USERNAME\AppData\Roaming\Microsoft\Internet Explorer\Quick Launch
\User Pinned\TaskBar\ChatGPT.Ink
startup: C:\ProgramData\Microsoft\Windows\Start Menu\Programs\ChatGPT
\ChatGPT.lnk

**Comment by ColtonHyland** [comment]
Mine as well

**Comment by decrisp1252** [comment]
Same here

**Comment by fruini** [comment]
Happened to me as well.

**Comment by pblue3** [comment]
Same here, should we be worried about it?

**Comment by 0com** [comment]
Me too

**Comment by hlatifpk** [comment]
I am also facing the same issue. ChatGPT was removed automatically from my system by Windows Defender. I reinstalled and Windows Defender instantly sent me notification which has following information.

> Trojan:Win64/Malgent!MSR
> Alert level: Severe
> Status: Active
> Date: 2/19/2023 4:31 AM
> Category: Trojan
> Details: This program is dangerous and executes commands from an attacker.
> Affected items: file: C:\Program Files\ChatGPT\ChatGPT.exe
> file: C:\ProgramData\Microsoft\Windows\Start Menu\Programs\ChatGPT\ChatGPT.lnk
> file: C:\Users\Public\Desktop\ChatGPT.lnk
> startup: C:\ProgramData\Microsoft\Windows\Start Menu\Programs\ChatGPT\ChatGPT.lnk

**Comment by jonasmarco** [comment]
Same here

**Comment by Ethkuil** [comment]
#436 Duplicate.
Current progress: https://github.com/lencx/ChatGPT/issues/436#issuecomment-1435723089

**Comment by RW2023** [comment]
Leaving a comment now but it did for me yesterday.

**Comment by JDH42** [comment]
![image](https://user-images.githubusercontent.com/12961024/219913171-15665a8a-7bc9-4b8d-a747-34fd40a471cb.png)

**Comment by JDH42** [comment]
Evening more concerning is that a secondary full scan found a persistent infection hiding inside the in Installer directory.

![image](https://user-images.githubusercontent.com/12961024/219913819-3c9d1ef8-3a8b-4262-ad24-2117296971dd.png)

**Comment by lencx** [comment]
> ![image](https://user-images.githubusercontent.com/12961024/219913171-15665a8a-7bc9-4b8d-a747-34fd40a471cb.png)

startup： It should be a boot auto-start feature that I added in a previous version, not a recent update. https://github.com/lencx/tauri-plugins-workspace/tree/dev/plugins/autostart

**Comment by 0xeb** [comment]
> 

it is open source project. feel free to audit it and build it yourself.

**Comment by ColtonHyland** [comment]
do the devs know why windows defender is flaggin this? I dont feel comfortable using the program after this flagged

**Comment by lencx** [comment]
> do the devs know why windows defender is flaggin this? I dont feel comfortable using the program after this flagged

I don't really know why it's flagged, but I'm sure it's safe. If you guys want to leave it, I respect your decision.

**Comment by lencx** [comment]
I've updated the readme. security agencies have now detected a trojan in the unknown download link.

**🛑 URGENT NOTICE: A hacker has been found to take advantage of the heat of `lencx/ChatGPT` to plant a Trojan horse after the fork project and rebuild the installer. If you have friends around you who are using this desktop application, please remind them not to download unknown links freely. Now the project will remove other installation ways and only provide this download link https://github.com/lencx/ChatGPT/releases.**

**🛑 紧急通知：目前发现有黑客利用 `lencx/ChatGPT` 的热度，在 fork 项目后植入木马，重新构建安装程序。如果你身边有朋友正在使用此桌面应用，请提醒 TA 们不要随意下载不明链接。现在项目将删除其他安装途径，仅提供此下载链接 https://github.com/lencx/ChatGPT/releases。**

**Comment by cycalo** [comment]
Can you explain the unknown download link ? I only ever used the known download link within this GitHub?

**Comment by lencx** [comment]
> Can you explain the unknown download link ? I only ever used the known download link within this GitHub?

I can't know the source of the dangerous download link at the moment either, because the project is too widespread at the moment, and it's open source, so anyone can rebuild it and release it. Once the application is flagged as a Trojan by windows defender, it seems that the application downloaded by this project will also be detected as a Trojan by security.

**Comment by lencx** [comment]
> Just wanted to suggest that Yara might be helpful in identifying potentially malicious forks of this project. Yara can detect malware based on patterns and signatures, so you could create Yara rules to describe the characteristics of the malware and then use Yara to scan the fork's code. Of course, scanning all forks can be time-consuming and it's important to respect users' privacy.
> 
> Yara is not a silver bullet, but it could be worth exploring as part of a comprehensive security strategy.

It may just be a clone of the repository code, not a fork.

**Comment by DemGiran** [comment]
> > Just wanted to suggest that Yara might be helpful in identifying potentially malicious forks of this project. Yara can detect malware based on patterns and signatures, so you could create Yara rules to describe the characteristics of the malware and then use Yara to scan the fork's code. Of course, scanning all forks can be time-consuming and it's important to respect users' privacy.
> > Yara is not a silver bullet, but it could be worth exploring as part of a comprehensive security strategy.
> 
> It may just be a clone of the repository code, not a fork.

what do you mean?

**Comment by olljanat** [comment]
@lencx two important thing which I see have have missing both here and your new project:
- You should digitally sign both binary and installer, at least on Windows. Currently also Edge browser warn about installed and suggest to removing it because of missing digital signature. Best option is if you can get official code signing certificate but even self signed would be better than nothing.
  - You can also send both of those for Microsoft from https://www.microsoft.com/en-us/wdsi/filesubmission/ so they will run automated tests for it and mark it to safe.
- You should modify those MSI packages in way that they can be installed also without admin rights to user profile.

**Comment by oliverw** [comment]
The installer should not need elevated permissions at all.

**Comment by olljanat** [comment]
Yes, that is fully doable like you can see from custom version https://github.com/olljanat/chatgpt

Here issue however is that only Tauri 2.0 can create that kind of installer and those are still on alpha versions.

**Comment by UBIadeWorks** [comment]
![imagen_2024-01-26_150528303](https://github.com/lencx/ChatGPT/assets/157633337/80f5c460-7334-4b3c-934e-b67abcc65bbb)
Got the warn trying to compile a Python program with pyinstaller

**Comment by AndreasGreenButterfly** [comment]
Same varning installed from Microsoft Store
